### PR TITLE
Support for Harvey. Harvey is Plan 9 with GPL.

### PR DESCRIPTION
**Pull request by rminnich**
Signed-off-by: Ronald G. Minnich <rminnich@gmail.com>

### CODE REVIEW

**Comment by rminnich** [comment]
If you think I don't know what I'm doing, you're right, but my pattern is learn by mistaking. :-)

File: src/librustc_target/spec/x86_64_harvey.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// This defines the amd64 target for the Linux Kernel. See the linux-kernel-base module for
```
Comment (oli-obk) [inline src/librustc_target/spec/x86_64_harvey.rs:1]: This comment needs some updating :laughing:

File: src/librustc_target/spec/x86_64_harvey.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    base.cpu = "x86-64".to_string();
+    base.max_atomic_width = Some(64);
+    base.features =
+        "-mmx,-sse,-sse2,-sse3,-ssse3,-sse4.1,-sse4.2,-3dnow,-3dnowa,-avx,-avx2,+soft-float"
+            .to_string();
+    base.code_model = Some(CodeModel::Kernel);
+    base.pre_link_args.get_mut(&LinkerFlavor::Gcc).unwrap().push("-m64".to_string());
+
+    Ok(Target {
+        // FIXME: Some dispute, the linux-on-clang folks think this should use "Linux"
```
Comment (oli-obk) [inline src/librustc_target/spec/x86_64_harvey.rs:17]: can you leave some links for further reading?

**Comment by oli-obk** [comment]
Does anyone have opinions on this @rust-lang/compiler ?

File: src/librustc_target/spec/x86_64_harvey.rs (hunk 1 above)
Comment (rminnich) [reply src/librustc_target/spec/x86_64_harvey.rs:1]: oops :-)

File: src/librustc_target/spec/x86_64_harvey.rs (hunk 2 above)
Comment (rminnich) [reply src/librustc_target/spec/x86_64_harvey.rs]: That was not my comment, so I removed it. I have no stake in this argument.

**Comment by rminnich** [comment]
Just FYI, once this is in, the sum total of system calls is 52, many of them being identical to Linux. When we started Harvey we made sure our system call convention followed the ABI, just to make compiler addition easy. Thanks for the super-fast review :-)

**Comment by Makimonus** [comment]
“Harvey is Plan 9 with GPL” except that it isn't 😛

**Comment by rminnich** [comment]
> “Harvey is Plan 9 with GPL” except that it isn't 😛

?

**Comment by rminnich** [comment]
ah I get it ... I mean Harvey moved to the the system v ABI system call interface, i.e. we use the sysv abi for x86_64, not the plan 9 one. The plan 9 one was stack-based, as still used in Go today.

**Comment by oli-obk** [comment]
I'll give everyone a bit more time to comment on this before merging. Could you leave some contact info (urls to repos or other related webpages) in the architecture files?

**Comment by nikomatsakis** [comment]
The main thing I would like, yes, is that I think we need some idea who to ping to notify of things related to this architecture. I don't have any objection to adding as a new Tier 3 platform.

**Comment by oli-obk** [comment]
@rminnich This PR needs rebasing and some contact info in case we need some help around the Harvey architecture. Then it can be merged.

**Comment by crlf0710** [comment]
@rminnich Ping from triage! This hasn't seen much activity recently. Could you address the comments above? Thanks!

**Comment by crlf0710** [comment]
@rminnich Ping from triage! I'm closing this due to inactivity. Feel free to reopen or create a new PR when you've got time to work on this again! Thanks!

**Pull request closed without merging**